image/tarexport/save.go

```diff
@@ -93,33 +105,68 @@ func (l *tarexporter) parseNames(names []string) (map[image.ID]*imageDescriptor,
 			if err != nil {
 				return nil, err
 			}
-			addAssoc(imgID, nil)
+			if err := addAssoc(imgID, nil); err != nil {
+				return nil, err
+			}
 			continue
 		}
 		if reference.IsNameOnly(namedRef) {
 			assocs := l.rs.ReferencesByName(namedRef)
 			for _, assoc := range assocs {
-				addAssoc(image.IDFromDigest(assoc.ID), assoc.Ref)
+				if err := addAssoc(image.IDFromDigest(assoc.ID), assoc.Ref); err != nil {
+					return nil, err
+				}
 			}
 			if len(assocs) == 0 {
 				imgID, err := l.is.Search(name)
 				if err != nil {
 					return nil, err
 				}
-				addAssoc(imgID, nil)
+				if err := addAssoc(imgID, nil); err != nil {
+					return nil, err
+				}
 			}
 			continue
 		}
 		id, err := l.rs.Get(namedRef)
 		if err != nil {
 			return nil, err
 		}
-		addAssoc(image.IDFromDigest(id), namedRef)
+		if err := addAssoc(image.IDFromDigest(id), namedRef); err != nil {
+			return nil, err
+		}
 
 	}
 	return imgDescr, nil
 }
 
+func (l *tarexporter) takeLayerReference(id image.ID, imgDescr *imageDescriptor) error {
+	img, err := l.is.Get(id)
+	if err != nil {
+		return err
+	}
+	imgDescr.image = img
+	if imgDescr.cachedLayer != nil {
```

Comment:
Sorry for late, We have 3 days holiday in China.

I thought if an image just have one empty layer, the `ChainID` may be emtpy string. But to check the code, it should be the empty layer ID. 

I think we can use `layerStore.Get` to Get an empty layer just like  [push](https://github.com/moby/moby/blob/master/distribution/push_v1.go#L217) operation.   So I think we do not need to check `nil` because of `EmptyLayer`. ( we can treat it as normal layer here)

But you mentioned that an image could have no layers and the ChainID should be empty.How does it happen? Could we use `imageStore.Get` to get this ? If couldn't, I think we should add a check here.

If I am wrong , correct me. Thanks.